src/integer_mod_q/poly_over_zq/from.rs

```diff
@@ -17,6 +17,44 @@ use crate::{error::MathError, integer::PolyOverZ, integer_mod_q::modulus::Modulu
 use flint_sys::fmpz_mod_poly::{fmpz_mod_poly_init, fmpz_mod_poly_set_fmpz_poly};
 use std::{mem::MaybeUninit, str::FromStr};
 
+impl From<(&PolyOverZ, &Modulus)> for PolyOverZq {
+    /// Create a [`PolyOverZq`] from a [`PolyOverZ`] and [`Modulus`].
+    ///
+    /// Parameters:
+    /// - `poly_modulus_tuple`: A tuple of the polynomial and the modulus.
+    ///
+    /// Examples:
+    /// ``` rust
+    /// use qfall_math::integer_mod_q::{PolyOverZq, Modulus};
+    /// use qfall_math::integer::PolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZ::from_str("4  0 1 102 3").unwrap();
+    /// let modulus = Modulus::from_str("100").unwrap();
+    ///
+    /// let mod_poly = PolyOverZq::from((&poly, &modulus));
+    ///
+    /// # let cmp_poly = PolyOverZq::from_str("4  0 1 2 3 mod 100").unwrap();
+    /// # assert_eq!(cmp_poly, mod_poly);
+    /// ```
+    fn from(poly_modulus_tuple: (&PolyOverZ, &Modulus)) -> Self {
+        let modulus = poly_modulus_tuple.1.clone();
+        let mut poly = MaybeUninit::uninit();
+        unsafe {
+            fmpz_mod_poly_init(poly.as_mut_ptr(), modulus.get_fmpz_mod_ctx_struct());
+            let mut poly = poly.assume_init();
```

Comment:
I feel like this can be put into a dedicated function returning a PolyOverZq, as this might be helpful in the future, improves modularity and allows to get something like a default value for PolyOverZq's